### PR TITLE
update cluster autoscaler to 1.21.0

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

**Review by ipochi — DISMISSED** [review]
lgtm

**Comment by knrt10** [comment]
LGTM but the CI though 😞

**Comment by surajssd** [comment]
@knrt10 don't worry. This upgrade is more work than it looks like. I have tried to carry many changes but for some reason the autoscaler pods are panicking. I haven't gotten to the bottom of it yet.

File: assets/charts/components/cluster-autoscaler/templates/secret.yaml [hunk 1]
```diff
@@ -16,21 +17,24 @@ data:
...
 type: Opaque
 stringData:
   cloud-config: |-
     [Global]
-    project-id= {{ .Values.packetProjectID }}
-    facility= {{ .Values.packetFacility }}
-    hostname-pattern={{ printf "%s" "{{.ClusterName}}-{{.NodeGroup}}-{{.RandString8}}" }}
-    os=flatcar_{{ .Values.packetOSChannel }}
+    project-id={{ .Values.packetProjectID }}
+    api-server-endpoint={{ .Values.apiServer }}
```
Comment (ipochi) [inline assets/charts/components/cluster-autoscaler/templates/secret.yaml:27]: What does this `api-server-endpoint` change do ?

can we minimize the order changes ?

File: assets/charts/components/cluster-autoscaler/Chart.yaml [hunk 2]
```diff
@@ -1,20 +1,19 @@
 apiVersion: v2
-appVersion: 1.18.1
+appVersion: 1.20.0
```
Comment (ipochi) [inline assets/charts/components/cluster-autoscaler/Chart.yaml:2]: We should update to 1.21.0

File: assets/charts/components/cluster-autoscaler/values.yaml [hunk 3]
```diff
@@ -93,13 +93,15 @@ azureNodeResourceGroup: ""
...
 # Required if cloudProvider=packet
+# XXX: Lokomotive specific change.
 packetClusterName: ""
 packetAuthToken: ""
 packetCloudInit: ""
 packetProjectID: ""
 packetFacility: ""
 packetOSChannel: ""
 packetNodeType: ""
+apiServer: ""
```
Comment (ipochi) [inline assets/charts/components/cluster-autoscaler/values.yaml:104]: If this is the new change can we have it in a separate commit, showing why ?

**Review by ipochi — DISMISSED** [review]
Nit and update the chart to 1.21.0.

File: assets/charts/components/cluster-autoscaler/templates/secret.yaml (hunk 1 above)
Comment (surajssd) [reply assets/charts/components/cluster-autoscaler/templates/secret.yaml:27]: I think I took it from [upstream](https://github.com/kubernetes/autoscaler/blob/5766bb71373a8f94faa0b8458b747cfbb89c1344/cluster-autoscaler/cloudprovider/packet/examples/cluster-autoscaler-secret.yaml#L25). I was trying to use the upstream configs to make this work with packet.

**Comment by ipochi** [comment]
@surajssd @invidian 

This is now updated to 1.21.0 and tested.

**Comment by ipochi** [comment]
This PR makes the use of kinvolk's image `repository: quay.io/kinvolk/cluster-autoscaler:fb8fdf8`.

This correspods to the [merge commit](https://github.com/kubernetes/autoscaler/pull/4274) in upstream kubernetes/autoscaler repo. 

The reason is that while the github.com/kubernetes/autoscaler/pull/4274 is merged, as of now newer version of autoscaler is not released. When a new version of autoscaler is released, the usage of this image must be stopped and reverted back to the upstream image.

**Comment by invidian** [comment]
@ipochi perhaps you can cherry-pick your merge commit to the branch starting with latest release tag and tag docker image using `git describe --tag --always`, so we can see that it's 3 commits on top of latest release, instead of picking all new code which has been merged?

**Comment by ipochi** [comment]
> This PR makes the use of kinvolk's image `repository: quay.io/kinvolk/cluster-autoscaler:fb8fdf8`.
> 
> This correspods to the [merge commit](https://github.com/kubernetes/autoscaler/pull/4274) in upstream kubernetes/autoscaler repo.
> 
> The reason is that while the github.com/[kubernetes/autoscaler/pull/4274](https://github.com/kubernetes/autoscaler/pull/4274) is merged, as of now newer version of autoscaler is not released. When a new version of autoscaler is released, the usage of this image must be stopped and reverted back to the upstream image.


Done.

Updated the image tag to `cluster-autoscaler-1.22.0-1-g5cd43fe7a`

File: assets/charts/components/cluster-autoscaler/templates/clusterrole.yaml [hunk 4]
```diff
@@ -52,6 +52,10 @@ rules:
       - replicationcontrollers
       - persistentvolumeclaims
       - persistentvolumes
+      ### Packet Specific (not available upstream)
```
Comment (invidian) [inline assets/charts/components/cluster-autoscaler/templates/clusterrole.yaml:55]: I think we should send those changes upstream.

**Review by invidian — COMMENTED** [review]
Looks good

**Review by knrt10 — APPROVED** [review]
+1

**Comment by surajssd** [comment]
I can't review this PR but LGTM :-)

**Comment by ipochi** [comment]
merging.

**Pull request merged**